ukoly/lekce_09_oop/creatures.py

```diff
@@ -0,0 +1,77 @@
+class Creature():
+    "Basic Creature class with no type and default stats"
+    def __init__(self, name, type = None):
+        self.name = name
+        self.type = type
+        self.damage = 5
+        self.hp = 10
+        self.defense = 0
+
+    def attack(self, creature):
+        if not(creature.is_alive()):
+            print("Cannot attack already dead creature ¯\\_(ツ)_/¯")
+        else:
+            real_damage = (self.damage - creature.defense)
+            if(real_damage > creature.hp):
+                real_damage = creature.hp
+                creature.hp = 0
+            else:
+                creature.hp -= real_damage
+            print("< {} attacked {} with {} damage ({} HP left (⋟﹏⋞) ) >".format(
+                self.name, creature.name, real_damage, creature.hp
+            ))
+            if not(creature.is_alive()):
+                print("< {} has died ✝>".format(creature.name))
+        
+
+    def is_alive(self):
+        return (self.hp > 0)
+
+    def __str__(self):
+        if(self.is_alive()):
+            return ("----------------------\n| Name: {}\n| Type: {}  \n|\n| Damage: {}\n| Defense: {}\n| HP: {}".format(self.name, self.type, self.damage, self.defense, self.hp))
```

Comment:
Vim, ze je spatne mit moc dlouhy radek kodu, ale nikde jsem nenasla, jak udelat nekolikaradkovy return statement a aby mi to fungovalo. Nejake navrhy? :) @rusinsky-pavel

Reply:
https://stackoverflow.com/questions/10660435/pythonic-way-to-create-a-long-multi-line-string

můžeš mrknout třeba sem :)

Nějakým způsobem si uložit ten string jako variable a tu pak dát do returnu -> výsledek stejný, ale asi by to bylo přehlednější

Reply:
Omg, jasny :woman_facepalming:
Diky :)